eng/mgmt/automation/generate_data.py

```diff
@@ -94,54 +97,116 @@ def generate(
+
+        command = 'autorest --version={0} --use={1} ' \
+                  '--java.azure-libraries-for-java-folder={2} --java.output-folder={3} ' \
+                  '--java.namespace={4} {5}'\
+            .format(
+                autorest,
+                use,
+                os.path.abspath(sdk_root),
+                os.path.abspath(output_dir),
+                namespace,
+                ' '.join((LLC_ARGUMENTS, input_arguments, credential_arguments, artifact_arguments, autorest_options))
+            )
+        logging.info(command)
+        if os.system(command) != 0:
+            logging.error('[GENERATE] Autorest fail')
+            return False
+
+        set_or_increase_version(sdk_root, GROUP_ID, module)
+        update_service_ci_and_pom(sdk_root, service, GROUP_ID, module)
+        update_root_pom(sdk_root, service)
+        update_version(sdk_root, output_dir)
 
     return True
 
 
-def install_build_tools(sdk_root: str):
-    command = 'mvn --no-transfer-progress clean install -f {0} -pl com.azure:sdk-build-tools'.format(os.path.join(sdk_root, 'pom.xml'))
+def compile_package(sdk_root: str, group_id: str, module: str):
+    command = 'mvn --no-transfer-progress clean verify package -f {0}/pom.xml -pl {1}:{2} -am'.format(
+        sdk_root, group_id, module)
     logging.info(command)
     if os.system(command) != 0:
-        logging.error('[COMPILE] Maven build fail for sdk-build-tools')
+        logging.error('[COMPILE] Maven build fail')
         return False
     return True
 
 
-def compile_package(output_dir: str):
-    command = 'mvn --no-transfer-progress clean verify package -f {0}'.format(os.path.join(output_dir, 'pom.xml'))
-    logging.info(command)
-    if os.system(command) != 0:
-        logging.error('[COMPILE] Maven build fail')
-        return False
-    return True
+def update_readme(output_dir: str, input_file: str, credential_types: str, credential_scopes: str, title: str) -> str:
+    readme_relative_path = ''
+
+    swagger_dir = os.path.join(output_dir, 'swagger')
+    if os.path.isdir(swagger_dir):
+        for filename in os.listdir(swagger_dir):
+            if filename.lower().startswith('readme') and filename.lower().endswith('.md'):
+                readme_yaml_found = False
+                readme_path = os.path.join(swagger_dir, filename)
+                with open(readme_path, 'r', encoding='utf-8') as f_in:
+                    content = f_in.read()
+                if content:
+                    yaml_blocks = re.findall(r'```\s?(?:yaml|YAML)\n(.*?)```', content, re.DOTALL)
+                    for yaml_str in yaml_blocks:
+                        yaml_json = yaml.safe_load(yaml_str)
+                        if 'low-level-client' in yaml_json and yaml_json['low-level-client']:
+                            # yaml block found, update
+                            yaml_json['input-file'] = [input_file]
+                            if title:
```

Comment:
title, credential_types and credential_types are the only three types we support for update?

Reply:
And `input-file`. Yes, so far.

Change service or module will result in a different folder.